clients/src/main/java/org/apache/kafka/clients/producer/internals/BuiltInPartitioner.java

```diff
@@ -40,8 +41,10 @@ public class BuiltInPartitioner {
     private final Logger log;
     private final String topic;
     private final int stickyBatchSize;
+    private final boolean rackAware;
+    private final String rack;
 
-    private volatile PartitionLoadStats partitionLoadStats = null;
+    private volatile PartitionLoadStatsHolder partitionLoadStats = null;
```

Comment:
```suggestion
    private volatile PartitionLoadStatsHolder partitionLoadStatsHolder = null;
```

As it got a bit confusing when later the code is retrieving `partitionLoadStats.inThisRack` which is a member of `PartitionLoadStatsHolder` of type `PartitionLoadStats`.  :D